app/components/thumbnail_component.rb

```diff
@@ -3,26 +3,47 @@
 class ThumbnailComponent < ApplicationComponent
   attr_reader :resource, :featured
 
-  # @param [SolrDocument, WorkVersion, Collection] resource
+  # @resource could be a SolrDocument or SolrDocumentAdapterDecorator (containing a Collection or Work),
+  # Collection, Work, WorkVersion, CollectionDecorator, WorkDecorator, or WorkVersionDecorator
   def initialize(resource:, featured: false)
     @resource = resource
     @featured = featured
   end
 
-  def icon
-    icon_map.fetch(icon_key, 'bar_chart')
+  def thumbnail
+    if thumbnail_url.present?
+      "<img src=#{thumbnail_url}></img>"
+    else
+      "<i class='material-icons material-icons--thumbnail'>#{icon}</i>"
+    end
   end
 
   def html_classes
     if featured?
-      'thumbnail col-xxl-6 ft-work__img'
+      if thumbnail_url.present?
+        'thumbnail-image col-xxl-6 ft-work__img'
+      else
+        'thumbnail-icon col-xxl-6 ft-work__img'
+      end
     else
-      'thumbnail'
+      if thumbnail_url.present?
+        'thumbnail-image'
+      else
+        'thumbnail-icon'
+      end
```

Comment:
I think you can avoid this nesting and just do:

```
if featured?
   # do stuff
else if thumbnail_url.present?
  'thumbnail-image'
else
  'thumbnail-icon'
end
```

Then you can remove the rubocop `Style/IfInsideElse` rule.

Reply:
Alternatively you could build up an array of classes based on `thumbnail_url.present?` and `featured?` and join them at the end, which may reduce the cognitive complexity a bit.